odc_eks/main.tf

```diff
@@ -53,17 +63,39 @@ module "vpc" {
   enable_nat_gateway           = var.enable_nat_gateway
   create_igw                   = var.create_igw
   create_database_subnet_group = true
-  enable_s3_endpoint           = var.enable_s3_endpoint
 
-  tags = merge(
-    {
-      Name        = "${local.cluster_id}-vpc"
-      owner       = var.owner
-      namespace   = var.namespace
-      environment = var.environment
-    },
-    var.tags
-  )
+  manage_default_security_group = false
+  manage_default_network_acl    = false
+  manage_default_route_table    = false
+
+  tags = local.tags
+}
+
+moved {
+  from = module.vpc[0].aws_vpc_endpoint.s3[0]
+  to   =  module.vpc_endpoints[0].aws_vpc_endpoint.this["s3"]
+}
+
+module "vpc_endpoints" {
+  source = "git::https://github.com/terraform-aws-modules/terraform-aws-vpc.git//modules/vpc-endpoints?ref=v5.1.1"
+  count  = var.create_vpc && var.enable_s3_endpoint ? 1 : 0
+
+  vpc_id = module.vpc[0].vpc_id
+  security_group_ids = [ module.vpc[0].default_security_group_id ]
+
+  endpoints = {
+    s3 = {
+      service      = "s3"
+      service_type = "Gateway"
+
+      route_table_ids = flatten([
+        module.vpc[0].private_route_table_ids,
+        module.vpc[0].public_route_table_ids
+      ])
+
+      tags = local.tags
+    }
+  }
 }
```

Comment:
The module has a `tags` var we should be passing `local.tags` to. I am not yet sure if that would mean the `tags` don't need to be passed to the `endpoints` block though, would need to test.

Reply:
Yes, I think if we pass `local.tags` as input to module `terraform-aws-vpc.git//modules/vpc-endpoints`, then we can stop passing it as part of `endpoints`. See https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/f7f84f70ad6db36e4eae0273bf08ed2ffd0a69c4/modules/vpc-endpoints/main.tf#L46.